src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -0,0 +1,430 @@
+package org.broadinstitute.hellbender.tools.walkers.variantutils;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.*;
+import org.broadinstitute.barclay.argparser.*;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.*;
+import org.broadinstitute.hellbender.cmdline.argumentcollections.DbsnpArgumentCollection;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.tools.walkers.GenotypeGVCFs;
+import org.broadinstitute.hellbender.tools.walkers.annotator.*;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.*;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.afcalc.FixedAFCalculatorProvider;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerArgumentCollection;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerGenotypingEngine;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.ReferenceConfidenceMode;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.genotyper.IndexedSampleList;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+import org.broadinstitute.hellbender.utils.variant.writers.GVCFWriter;
+import picard.cmdline.programgroups.OtherProgramGroup;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * Condense homRef blocks in a single-sample GVCF
+ *
+ * <p>
+ * ReblockGVCF compressed a GVCF by merging hom-ref blocks that were produced using the '-ERC GVCF' or '-ERC BP_RESOLUTION' mode of the
+ * HaplotypeCaller according to new GQ band parameters.  A joint callset produced with GVCFs reprocessed by ReblockGVCF will have
+ * lower precision for hom-ref genotype qualities at variant sites, but the input data footprint can be greatly reduced
+ * if the default GQ band parameters are used.</p>
+ *
+ * <h3>Input</h3>
+ * <p>
+ * A HaplotypeCaller-produced gVCF to reblock
+ * </p>
+ *
+ * <h3>Output</h3>
+ * <p>
+ * A smaller GVCF.
+ * </p>
+ *
+ * <h3>Usage example</h3>
+ * <pre>
+ * gatk ReblockGVCF \
+ *   -R reference.fasta \
+ *   -V sample1.g.vcf \
+ *   -O sample1.reblocked.g.vcf
+ * </pre>
+ *
+ * <h3>Caveats</h3>
+ * <p>Only single-sample gVCF files produced by HaplotypeCaller can be used as input for this tool.</p>
+ * <p>By default this tool only passes through annotations used by VQSR.  A different set of annotations can be specified with the usual -A argument.  Annotation groups are not supported by this tool.</p>
```

Comment:
I think annotation groups are automatically supported and propagated to the annotation engine via `makeVariantAnnotations`, unless the tool manually intervenes.

Reply:
I agree -- that was probably just a leftover from the GATK3 version.